src/server/lib/oauth/callback.js

```diff
@@ -253,12 +253,20 @@ async function _getOAuthAccessToken (code, provider, callback) {
 
 // Ported from https://github.com/ciaranj/node-oauth/blob/a7f8a1e21c362eb4ed2039431fb9ac2ae749f26a/lib/oauth2.js
 function _get (provider, accessToken, callback) {
-  const url = provider.profileUrl
+  let url = provider.profileUrl
   const headers = provider.headers || {}
+  const setGetAccessTokenProfileUrl = provider.setGetAccessTokenProfileUrl !== null ? provider.setGetAccessTokenProfileUrl : true
 
   if (this._useAuthorizationHeaderForGET) {
     headers.Authorization = this.buildAuthHeader(accessToken)
 
+    // Mail.ru requires 'access_token' as URL request parameter
+    if (setGetAccessTokenProfileUrl) {
+      if (url) {
+        url = url + '?access_token=' + accessToken
```

Comment:
I am afraid append url like this could be dangerous. What if the origin profileUrl contains '?' already?

Suggest to use Node.js url.searchParams.append to handle it.